File: azkar/src/Core/Startup.swift
// Starting at login, and whether the window opens when the app starts.
import Foundation

/// The LaunchAgent that starts the app at login, in the background, and restarts it if it crashes.
func launchAgentPlist(label: String, executable: String) -> Data {
  let plist: [String: Any] = [
    "Label": label,
    "ProgramArguments": [executable, "--background"],
    "RunAtLoad": true,
    "KeepAlive": ["SuccessfulExit": false],
    "ProcessType": "Interactive",
  ]
  return try! PropertyListSerialization.data(fromPropertyList: plist, format: .xml, options: 0)
}

/// Opened by hand (Finder, Spotlight, Dock): always. Started at login (`--background`): only if asked.
func opensWindowAtLaunch(arguments: [String], config: Config) -> Bool {
  !arguments.contains("--background") || config.showWindowAtLogin
}

File: azkar/kotlin/shared/src/commonMain/kotlin/com/wildduck/azkar/core/Startup.kt
// Whether the window opens when the app starts. (The macOS LaunchAgent plist stays in Swift.)
package com.wildduck.azkar.core

/** Opened by hand: always. Started at login (`--background`): only if asked. */
fun opensWindowAtLaunch(arguments: List<String>, config: Config): Boolean =
    "--background" !in arguments || config.showWindowAtLogin
